### PR TITLE
Issue 425 - Migrate to DotNet 8 - File-scoped namespaces: Algorithms,…

### PR DESCRIPTION
**Pull request by gmottajr**
Issue 425 - Migrate to DotNet 8 - File-scoped namespaces: Algorithms, Crypto, DataCompression, Encoders, Graph

Please let me know whether or not it would be necessary to create new unit tests for this change.

Motivation and Context
Github issue https://github.com/TheAlgorithms/C-Sharp/issues/425: Proposed to  update to File-scoped namespaces. It was requested to perform the changes in a few PRs to make them smaller and easier to review. This is the one only meant to update .NET version.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Comments in areas I changed are up to date
- [ ] I have added comments to hard-to-understand areas of my code
- [ ] I have made corresponding changes to the README.md

### CODE REVIEW

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by gmottajr** [comment]
I'm concerning whether or not you would like a unit test for changes similar to this one. I can create them, just not sure if that is something that you think worth doing.
Another thing is related to the DotNet 8 upgrate, we could create a test that would check all project files associated to that solution if they are targeted to DotNet 8 or higher. This way I could require that any project that eventually gets added to that solution will be at least targeted .Net8. How does that sound?
Please let me know your thoughts.

**Comment by siriak** [comment]
@gmottajr, in general, it's a good idea to have architectural tests. But for targeting a version of .NET there are better ways. Firstly, we don't add new projects very often (we barely ever add them) and it's easy to see the version just by opening the project file. Secondly, we could add `Directory.Build.props` file to specify all versions in one place, but I think it would complicate the project for new developers who might be unfamiliar with it. Besides, we will need a new project just for these tests because they are neither algorithm tests nor data structures tests.

To sum up, I think we don't need such tests here because it would unnecessarily complicate the project for new developers.

**Comment by gmottajr** [comment]
Sure, that makes sense. 
So never mind.
